GNUmakefile

```diff
@@ -29,45 +30,40 @@ subnet:
 	@sh -c "'$(CURDIR)/scripts/setup_test_subnet.sh'"
 
 # test runs the test suite
-test: subnet generate deps
-	go list $(TEST) | xargs -n1 go test $(TESTARGS)
+test: subnet generate
+	go list $(TEST) | grep -v ^github.com/hashicorp/serf/vendor | xargs -n1 go test $(TESTARGS)
 
 # testrace runs the race checker
-testrace: subnet generate deps
+testrace: subnet generate
 	go test -race $(TEST) $(TESTARGS)
 
-# deps installs all the dependencies needed to test, build, and run
-deps:
-	go get github.com/mitchellh/gox
-	go get -v -d -t ./...
-
-# `go get -u` causes git to revert Serf to the master branch. This causes all
-# kinds of headaches. We record the git sha when make starts try to correct it
-# if we detect dift. DO NOT use `git checkout -f` for this because it will wipe
-# out your changes without asking.
-updatedeps:
-	@echo "INFO: Currently on $(GITBRANCH) ($(GITSHA))"
-	@git diff-index --quiet HEAD ; if [ $$? -ne 0 ]; then \
-		echo "ERROR: Your git working tree has uncommitted changes. updatedeps will fail. Please stash or commit your changes first."; \
-		exit 1; \
-	fi
-	go get -u github.com/mitchellh/gox
-	go list ./... \
-		| xargs go list -f '{{join .Deps "\n"}}' \
-		| grep -v github.com/hashicorp/serf \
-		| grep -v '/internal/' \
-		| sort -u \
-		| xargs go get -f -u -v -d
-	@if [ "$(GITBRANCH)" != "" ]; then git checkout -q $(GITBRANCH); else git checkout -q $(GITSHA); fi
-	@if [ `git rev-parse HEAD` != "$(GITSHA)" ]; then \
-		echo "ERROR: git checkout has drifted and we weren't able to correct it. Was $(GITBRANCH) ($(GITSHA))"; \
-		exit 1; \
-	fi
-	@echo "INFO: Currently on $(GITBRANCH) ($(GITSHA))"
+# updatedeps installs all the dependencies needed to test, build, and run
+updatedeps:: tools
+	govendor list -no-status +vendor | xargs -n1 go get -u
+	govendor update +vendor
 
 # generate runs `go generate` to build the dynamically generated source files
```

Comment:
I couldn't find any actual generate stuff in Consul so this is worth checking and deleting if that's the case for Serf, too, especially since this is a parent of everything.